java-plugin/src/main/java/fr/greencodeinitiative/java/checks/CookieWithoutExpirationRule.java

```diff
@@ -0,0 +1,100 @@
+package fr.greencodeinitiative.java.checks;
+import java.util.*;
+
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.model.declaration.VariableTreeImpl;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.*;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+
+
+@Rule(
+        key = "CRJVM207-JAVA",
+        name = "Developpement",
+        description = CookieWithoutExpirationRule.MESSAGERULE,
+        priority = Priority.MINOR,
+        tags = {"smell"})
+
+public class CookieWithoutExpirationRule extends IssuableSubscriptionVisitor {
+
+    protected static final String MESSAGERULE = "Customer data must have end-of-life information, so cookies must have a maxAge";
+
+
+ @Override
+ public List<Tree.Kind> nodesToVisit() {
+     return Collections.singletonList(Tree.Kind.CLASS);
+ }
+ private final CookieWithoutExpirationRuleCheckVisitor visitorInFile = new CookieWithoutExpirationRuleCheckVisitor();
+
+
+ @Override
+ public void visitNode(Tree tree) {
+
+     tree.accept(visitorInFile);
+     //Class visitor
+     if (visitorInFile.hasANewCookieWithoutMaxDate())
+     {
+         //if we found a cookie that maxDate is not initialized, we report issue
+         reportIssue(tree, "Avoid not setting MaxAge");
+     }
+ }
+
+    private class CookieWithoutExpirationRuleCheckVisitor extends BaseTreeVisitor {
+
+
+        // storage of variable name for setMaxAge
+        private ArrayList<String> hasSetMaxAgeForCookiesVariableName = new ArrayList<>();
+        // storage of variable name for New Cookies
+        private ArrayList<String>  newCookieVariableName = new ArrayList<>();
+        @Override
+        public void visitReturnStatement(ReturnStatementTree tree) {
+            this.scan(tree.expression());
+        }
+
+        @Override
+        public void visitVariable(VariableTree tree) {
+            //when we visit variable affectation
+            for (Tree children : ((VariableTreeImpl) tree).children())
+            {
+                //if we found an affectation
+                if (children.is(Tree.Kind.NEW_CLASS)
+                        && ((IdentifierTree)((NewClassTree)children).identifier()).toString().equals("Cookie"))
+                {
+                    //if this is a New Cookie affectation, we store the name of the variable
+                    this.newCookieVariableName.add(tree.simpleName().toString());
+                }
+                else
+                    super.visitVariable(tree);
+            }
+        }
+
+
+        public boolean hasANewCookieWithoutMaxDate()
```

Comment:
checkin no SetMaxAge, I agree. but I think we also can verify that the maxAge isn't so far in the futur (like 2 years or more for example).

Reply:
Hello, 
We tough about This durring the hackathon and we came to conclusion, checking maxage value should be an other rule about the expiration strategy.

Reply:
maybe to say it in description issue ?